CRM/Utils/Check/Component/Case.php

```diff
@@ -397,4 +397,114 @@ public function checkRelationshipTypeProblems() {
+      }
+
+      if (empty($case_type['definition']['caseRoles'])) {
+        $messages[] = new CRM_Utils_Check_Message(
+          __FUNCTION__ . "missingcaseroles",
+          '<p>' . ts('CaseRoles seems to be missing in the xml file for Case Type "<em>%1</em>".',
+          [
+            1 => htmlspecialchars(empty($case_type['title']) ? $dao->id : $case_type['title']),
+          ]) . '</p>',
+          ts('Missing Case Roles'),
+          \Psr\Log\LogLevel::ERROR,
+          'fa-exclamation'
+        );
+        continue;
+      }
+
+      // Loop thru each role in the xml.
+      foreach ($case_type['definition']['caseRoles'] as $role) {
+        $name_to_suggest = NULL;
+        $xml_name = $role['name'];
+        if (isset($names_a_b[$xml_name]) || isset($names_b_a[$xml_name])) {
+          // It matches a name, so either name and label are the same or it's
+          // an edge case already dealt with by core status checks, so do
+          // nothing.
+          continue;
+        }
+        elseif (isset($labels_b_a[$xml_name])) {
+          // $labels_b_a[$xml_name] gives us the id, so then look up name_b_a
+          // from the original relationship_types array which is keyed on id.
+          // We do b_a first because it's the more standard one, although it
+          // will only make a difference in edge cases which we leave to the
+          // other checks.
+          $name_to_suggest = $relationship_types[$labels_b_a[$xml_name]]['name_b_a'];
+        }
+        elseif (isset($labels_a_b[$xml_name])) {
+          $name_to_suggest = $relationship_types[$labels_a_b[$xml_name]]['name_a_b'];
+        }
+
+        // If it didn't match any name or label then that's weird.
+        if (empty($name_to_suggest)) {
+          $messages[] = new CRM_Utils_Check_Message(
+            __FUNCTION__ . "invalidcaserole",
+            '<p>' . ts('CaseRole "<em>%1</em>" in the xml file for Case Type "<em>%2</em>" doesn\'t seem to match any existing relationship type.',
+            [
+              1 => htmlspecialchars($xml_name),
+              2 => htmlspecialchars(empty($case_type['title']) ? $dao->id : $case_type['title']),
+            ]) . '</p>',
+            ts('Invalid Case Role'),
+            \Psr\Log\LogLevel::ERROR,
+            'fa-exclamation'
+          );
+        }
+        else {
+          $messages[] = new CRM_Utils_Check_Message(
+            __FUNCTION__ . "suggestedchange",
+            '<p>' . ts('Suggest changing Case Role "<em>%1</em>" in the xml file for Case Type "<em>%2</em>" to "<em>%3</em>". Using label is deprecated as of version 5.20.',
+            [
+              1 => htmlspecialchars($xml_name),
+              2 => htmlspecialchars(empty($case_type['title']) ? $dao->id : $case_type['title']),
+              3 => htmlspecialchars($name_to_suggest),
+            ]) . '</p>',
+            ts('Case Role using display label instead of internal machine name'),
+            \Psr\Log\LogLevel::NOTICE,
```

Comment:
```suggestion
            \Psr\Log\LogLevel::WARNING,
```
Per [PSR-3](https://www.php-fig.org/psr/psr-3/) this should be a warning because it is actually undesirable, not merely significant.

Reply:
I was basing this on the existing one about portable resource urls. It's "everything is working right now, but consider changing this to make it more resilient". But yes I see from the logging spec it is more than something like an "interesting event".